src/api/api_test.go

```diff
@@ -249,12 +249,8 @@ func (s testServer) Cleanup() {
 }
 
 func (s testServer) client(addr string) testClient {
-	if addr == "" {
-		addr = s.GetAddr()
-	} else {
-		_, port, _ := net.SplitHostPort(s.GetAddr())
-		addr = net.JoinHostPort(addr, port)
-	}
+	_, port, _ := net.SplitHostPort(s.GetAddr())
+	addr = net.JoinHostPort(addr, port)
```

Comment:
None of our tests pass empty string as the client address, so I removed the conditional for this.